### PR TITLE
Add docs for kotlin codegen

### PR DESCRIPTION
**Pull request by mbossenbroek**

### CODE REVIEW

File: docs/advanced/kotlin-codegen.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# About
+
+In addition to the Java code-gen, we also have an experimental Kotlin API that generates more idiomatic Kotlin classes. Instead of generating Java code that can be used in Kotlin, it generates native Kotlin code that leans into techniques that only Kotlin supports. This is split primarily into two parts: the data classes used to serialize responses (both in the server and in the client), and the client query projections
```
Comment (srinivasankavitha) [inline docs/advanced/kotlin-codegen.md:3]: Would it help to also clarify that you can generate kotlin classes using the regular approach (linking to it). Currently you mention Java codegen, btut would be helpful to clarify that it also generates kotlin. What you describe here is a more idiomatic apprach etc.

Comment (srinivasankavitha) [reply docs/advanced/kotlin-codegen.md:3]: The docs look great, thanks for adding!

File: docs/advanced/kotlin-codegen.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+```kotlin
+val title: String = series.title
+val releaseDate: Int = series.releaseDate
+val endDate: Int? = series.endDate
+```
+
+## Query Projections
+
+Kotlin supports [function literals with a receiver](https://kotlinlang.org/docs/lambdas.html#function-literals-with-receiver) and this allows us to mimic the GQL query syntax, directly in the language. An advantage of writing queries directly in the language is that when the schema changes, any incompatibilities will show up as compile time errors, and intellisense can guide users to craft queries. Additionally, nested projections are nested in the query, just as they appear in a GQL query. For example:
```
Comment (paulbakker) [inline docs/advanced/kotlin-codegen.md:50]: Don't use "intellisense"; make it generic like "your IDE"

File: docs/advanced/kotlin-codegen.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
+# About
```
Comment (paulbakker) [inline docs/advanced/kotlin-codegen.md:1]: For better PR reviewing in the future, I recommend putting each sentence on a new line. The rendering will be the same, but you can more specifically put a comment on a line.

File: docs/advanced/kotlin-codegen.md (hunk 1 above)
Comment (paulbakker) [inline docs/advanced/kotlin-codegen.md:3]: It's called "experimental" here. What is your plan to get it out of an experimental phase?

**Review by paulbakker — CHANGES_REQUESTED** [review]

File: docs/advanced/kotlin-codegen.md (hunk 1 above)
Comment (mbossenbroek) [reply docs/advanced/kotlin-codegen.md:3]: That's a great question - it's perfectly stable & we've been using it for some time now with no issues. If you have no objections then I'd be happy dropping the label 🤷‍♂️

File: docs/advanced/kotlin-codegen.md (hunk 2 above)
Comment (mbossenbroek) [reply docs/advanced/kotlin-codegen.md:50]: huh - I had no idea that was a msft thing. Guess I must have picked it up there & have been using it for any IDE 🤔

File: docs/advanced/kotlin-codegen.md (hunk 1 above)
Comment (mbossenbroek) [reply docs/advanced/kotlin-codegen.md]: Updated

File: docs/advanced/kotlin-codegen.md (hunk 2 above)
Comment (mbossenbroek) [reply docs/advanced/kotlin-codegen.md]: Fixed

File: docs/advanced/kotlin-codegen.md (hunk 3 above)
Comment (mbossenbroek) [reply docs/advanced/kotlin-codegen.md:1]: Done

**Review by srinivasankavitha — APPROVED** [review]

**Review by paulbakker — APPROVED** [review]

**Pull request merged**